server/oneoff/database_setup.py

```diff
@@ -530,106 +533,130 @@
-                                    i
-                                ]["minStat"],
-                                max_damage=level["normalEffects"]["modifiableEffect"][
-                                    i
-                                ]["maxStat"],
+                        for locale in spell["name"]:
+                            spell_translation = ModelSpellTranslation(
+                                spell_id=spell_object.uuid,
+                                locale=locale,
+                                name=spell["name"][locale],
+                                description=spell["description"][locale],
                             )
+                            db_session.add(spell_translation)
+                            spell_object.spell_translation.append(spell_translation)
 
-                            if level["criticalEffects"].get("modifiableEffect", None):
-                                spell_effect.crit_min_damage = level["criticalEffects"][
-                                    "modifiableEffect"
-                                ][i]["minStat"]
-                                spell_effect.crit_max_damage = level["criticalEffects"][
-                                    "modifiableEffect"
-                                ][i]["maxStat"]
+                        create_spell_stats(db_session, spell, spell_object)
 
-                            db.session.add(spell_effect)
-                            spell_stat.spell_effects.append(spell_effect)
 
-                        db.session.add(spell_stat)
-                        spell_object.spell_stats.append(spell_stat)
-
-                    db.session.add(spell_object)
-                    spell_pair_object.spells.append(spell_object)
-
-                db.session.add(spell_pair_object)
-                class_object.spell_variant_pairs.append(spell_pair_object)
+if __name__ == "__main__":
+    # print("Resetting database")
+    # base.Base.metadata.reflect(base.engine)
+    # base.Base.metadata.drop_all(base.engine)
+    # base.Base.metadata.create_all(base.engine)
+    redis_connection.flushall()
 
-        db.session.commit()
+    while True:
+        response = input("Would you like to populate all tables (y/n)? ")
+        if response == "y":
+            add_item_types_and_slots()
+            add_sets_and_items()
+            add_weapons()
+            add_pets()
+            add_mounts()
+            add_classes_and_spells()
+            break
+        elif response == "n":
+            add_item_types_and_slots()
+
+            while True:
+                response = input("Would you like to add sets and items? (y/n)? ")
+                if response == "y":
+                    add_sets_and_items()
+                    break
+                elif response == "n":
+                    break
+                else:
+                    print("Invalid response, please type 'y' or 'n'")

```

Comment:
Can you rewrite this into a function that takes in the input string and the `add_*` function you want to call so you can reduce the amount of duplicated code?